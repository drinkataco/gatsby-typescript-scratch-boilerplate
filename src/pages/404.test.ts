import { useStaticQuery, StaticQuery } from 'gatsby';

import NotFoundPage from './404';

describe('not found page', () => {
  beforeEach(() => {
    (StaticQuery as jest.Mock).mockImplementationOnce(({ render }) => render({}));
    (useStaticQuery as jest.Mock).mockReturnValue({
      allMarkdownRemark: {
        nodes: [
          {
            fields: { slug: 'this-page' },
            frontmatter: { title: 'A Page #1' },
          },
          {
            fields: { slug: 'this-page' },
            frontmatter: { title: 'A Page #2' },
          },
        ],
      },
    });
  });

  it('renders 404 page', () => {
    expect.assertions(1);

    expect(NotFoundPage()).toMatchSnapshot();
  });
});
