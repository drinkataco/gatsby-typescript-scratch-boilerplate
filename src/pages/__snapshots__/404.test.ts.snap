// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not found page renders 404 page 1`] = `
<MainLayout>
  <h1
    className="font-bold mb-6"
  >
    404 Not Found
  </h1>
  <h3>
    Why don't you try...
  </h3>
  <ul>
    <li>
      <mockConstructor
        to="this-page"
      >
        A Page #1
      </mockConstructor>
    </li>
    <li>
      <mockConstructor
        to="this-page"
      >
        A Page #2
      </mockConstructor>
    </li>
  </ul>
</MainLayout>
`;
